src/kubernetes.cc

```diff
@@ -907,6 +976,80 @@ json::value KubernetesReader::FindTopLevelController(
   return FindTopLevelController(ns, GetOwner(ns, controller_ref));
 }
 
+void KubernetesReader::UpdateServiceToMetadataCache(
+    const json::Object* service, bool is_deleted) throw(json::Exception) {
+#ifdef VERBOSE
+  LOG(DEBUG) << "UpdateServiceToMetadataCache(" << *service << ")";
+#endif
+  const json::Object* metadata = service->Get<json::Object>("metadata");
+  const std::string namespace_name = metadata->Get<json::String>("namespace");
+  const std::string service_name = metadata->Get<json::String>("name");
+  const ServiceKey service_key(namespace_name, service_name);
+
+  std::lock_guard<std::mutex> lock(service_mutex_);
+  if (is_deleted) {
+    service_to_metadata_.erase(service_key);
+  } else {
+    auto it_inserted =
+        service_to_metadata_.emplace(service_key, json::value());
+    it_inserted.first->second = service->Clone();
+  }
+}
+
+void KubernetesReader::UpdateServiceToPodsCache(
+    const json::Object* endpoints, bool is_deleted) throw(json::Exception) {
```

Comment:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#endpoints-v1-core

The resource is called Endpoints, not Endpoint, since it lists all endpoints used by a single service.

Added comments to this method in the header file.